src/components/ContentHighlights/HighlightStepper/ContentHighlightStepper.jsx

```diff
@@ -0,0 +1,111 @@
+import React, {
+  useState, useEffect, useContext,
+} from 'react';
+import {
+  Stepper, FullscreenModal, Container, Button,
+} from '@edx/paragon';
+import PropTypes from 'prop-types';
+import HighlightStepperTitle from './HighlightStepperTitle';
+import HighlightStepperSelectCourses from './HighlightStepperSelectCourses';
+import HighlightStepperConfirmCourses from './HighlightStepperConfirmCourses';
+import HighlightStepperConfirmHighlight from './HighlightStepperConfirmHighlight';
+import HighlightStepperFooterHelpLink from './HighlightStepperFooterHelpLink';
+import { ContentHighlightsContext } from '../ContentHighlightsContext';
+/**
+ * Stepper Modal Currently accessible from:
+ *  - ContentHighlightSetCard
+ *  - CurrentContentHighlightHeader
+ *  - ZeroStateHighlights
+ *
+ * @param {object} args Arugments
+ * @param {boolean} args.isOpen Whether the modal containing the stepper is currently open.
+ * @returns
+ */
+const ContentHighlightStepper = ({ isOpen }) => {
+  const { setIsModalOpen } = useContext(ContentHighlightsContext);
+  /* eslint-disable no-unused-vars */
+  const steps = ['Title', 'Select courses', 'Confirm and Publish', 'All Set'];
+  const [currentStep, setCurrentStep] = useState(steps[0]);
+  const [modalState, setModalState] = useState(isOpen);
+  useEffect(() => {
+    setModalState(isOpen);
+  }, [isOpen]);
+  const submitAndReset = () => {
+    if (steps.indexOf(currentStep) === steps.length - 1) {
+      /* TODO: submit data to api if confirmed */
+      setCurrentStep(steps[0]);
+    }
+    setIsModalOpen(false);
+  };
+  return (
+    <>
+      <Stepper activeKey={currentStep}>
+        <FullscreenModal
+          title="New Highlight"
+          className="bg-light-200"
+          isOpen={modalState}
+          onClose={() => {
+            submitAndReset();
```

Comment:
Yes exactly, since it was just the first pass building the stepper shell. I used a very basic catchall function for now, but it would need to be extracted out into separate functions for the expected behavior.